dataline-workers/src/test/java/io/dataline/workers/singer/SingerTapFactoryTest.java

```diff
@@ -0,0 +1,164 @@
+import io.dataline.workers.DefaultSyncWorker;
+import io.dataline.workers.InvalidCredentialsException;
+import io.dataline.workers.JobStatus;
+import io.dataline.workers.OutputAndStatus;
+import io.dataline.workers.StreamFactory;
+import io.dataline.workers.TestConfigHelpers;
+import io.dataline.workers.WorkerUtils;
+import io.dataline.workers.process.ProcessBuilderFactory;
+import io.dataline.workers.protocol.singer.MessageUtils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SingerTapFactoryTest {
+  private static final String IMAGE_NAME = "hudi:latest";
+  private static final String JOB_ROOT_PREFIX = "workspace";
+  private static final String DISCOVER_SCHEMA_JOB_ROOT_PREFIX = "discover";
+  private static final String CONFIG_JSON_FILENAME = "tap_config.json";
+  private static final String CATALOG_JSON_FILENAME = "catalog.json";
+  private static final String STATE_JSON_FILENAME = "input_state.json";
+  private static final String TABLE_NAME = "user_preferences";
+  private static final String COLUMN_NAME = "favorite_color";
+
+  private Path jobRoot;
+  private Path discoverSchemaJobRoot;
+  private Path errorLogPath;
+
+  @BeforeEach
+  public void setup() {
+    try {
+      jobRoot = Files.createTempDirectory(JOB_ROOT_PREFIX);
+      discoverSchemaJobRoot = jobRoot.resolve(DISCOVER_SCHEMA_JOB_ROOT_PREFIX);
+      errorLogPath = jobRoot.resolve(DefaultSyncWorker.TAP_ERR_LOG);
+
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void test() throws InvalidCredentialsException, IOException {
+    StreamFactory streamFactory = mock(StreamFactory.class);
+    SingerMessage recordMessage1 =
+        MessageUtils.createRecordMessage(TABLE_NAME, COLUMN_NAME, "blue");
+    SingerMessage recordMessage2 =
+        MessageUtils.createRecordMessage(TABLE_NAME, COLUMN_NAME, "yellow");
+
+    final List<SingerMessage> expected = Lists.newArrayList(recordMessage1, recordMessage2);
+
+    StandardTapConfig tapConfig =
+        WorkerUtils.syncToTapConfig(TestConfigHelpers.createSyncConfig().getValue());
+    StandardDiscoverSchemaInput discoverSchemaInput = new StandardDiscoverSchemaInput();
+    discoverSchemaInput.setConnectionConfiguration(
+        tapConfig.getSourceConnectionImplementation().getConfiguration());
+    ProcessBuilderFactory pbf = mock(ProcessBuilderFactory.class);
+    ProcessBuilder processBuilder = mock(ProcessBuilder.class);
+    Process process = mock(Process.class);
+    InputStream inputStream = mock(InputStream.class);
+    SingerDiscoverSchemaWorker discoverSchemaWorker = mock(SingerDiscoverSchemaWorker.class);
+
+    final SingerColumn singerColumn = new SingerColumn();
```

Comment:
i'll move all of this into a helper on the next test suite as i think i will have to reuse it.